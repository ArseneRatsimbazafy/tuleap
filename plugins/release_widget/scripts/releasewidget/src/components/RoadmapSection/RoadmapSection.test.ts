/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount, Wrapper } from "@vue/test-utils";
import RoadmapSection from "./RoadmapSection.vue";
import { createStoreMock } from "@tuleap-vue-components/store-wrapper-jest";
import Vue from "vue";
import GetTextPlugin from "vue-gettext";
import { StoreOptions } from "../../type";

const project_id = 102;
function getPersonalWidgetInstance(store_options: StoreOptions): Wrapper<RoadmapSection> {
    const store = createStoreMock(store_options);
    const component_options = {
        mocks: { $store: store }
    };
    Vue.use(GetTextPlugin, {
        translations: {},
        silent: true
    });

    return shallowMount(RoadmapSection, component_options);
}

describe("RoadmapSection", () => {
    let store_options: StoreOptions;
    beforeEach(() => {
        store_options = {
            state: {
                is_loading: false,
                current_milestones: [],
                project_id: project_id
            },
            getters: {
                has_rest_error: false
            }
        };

        getPersonalWidgetInstance(store_options);
    });

    it("Given user display widget, Then a good link to top planning of the project is rendered", () => {
        const wrapper = getPersonalWidgetInstance(store_options);

        expect(wrapper.element).toMatchSnapshot();
    });
});
