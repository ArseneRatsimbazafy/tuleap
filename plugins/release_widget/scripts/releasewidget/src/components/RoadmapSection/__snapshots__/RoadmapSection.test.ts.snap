// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoadmapSection Given user display widget, Then a good link to top planning of the project is rendered 1`] = `
<div
  class="project-release-timeframe"
>
  <span
    class="project-release-label"
  >
    Roadmap
  </span>
   
  <div
    class="project-other-releases"
  >
    <div
      class="project-release-roadmap-icon"
    >
      <i
        class="fa fa-angle-double-up"
      />
    </div>
     
    <a
      class="releases-link"
      data-test="backlog-link"
      href="/plugins/agiledashboard/?action=show-top&group_id=102&pane=topplanning-v2"
    >
      <translate-stub
        tag="span"
        translateparams="[object Object]"
      >
        
                %{nb_backlog_items} items in the backlog, %{nb_upcoming_releases} upcoming releases
            
      </translate-stub>
    </a>
  </div>
</div>
`;
